js/plugins/google-genai/src/googleai/gemini.ts

```diff
@@ -0,0 +1,601 @@
+      mode: z.enum(['MODE_UNSPECIFIED', 'AUTO', 'ANY', 'NONE']).optional(),
+      allowedFunctionNames: z.array(z.string()).optional(),
+    })
+    .describe(
+      'Controls how the model uses the provided tools (function declarations). ' +
+        'With AUTO (Default) mode, the model decides whether to generate a ' +
+        'natural language response or suggest a function call based on the ' +
+        'prompt and context. With ANY, the model is constrained to always ' +
+        'predict a function call and guarantee function schema adherence. ' +
+        'With NONE, the model is prohibited from making function calls.'
+    )
+    .optional(),
+  responseModalities: z
+    .array(z.enum(['TEXT', 'IMAGE', 'AUDIO']))
+    .describe(
+      'The modalities to be used in response. Only supported for ' +
+        "'gemini-2.0-flash-exp' model at present."
+    )
+    .optional(),
+  googleSearchRetrieval: z
+    .union([z.boolean(), z.object({}).passthrough()])
+    .describe(
+      'Retrieve public web data for grounding, powered by Google Search.'
+    )
+    .optional(),
+}).passthrough();
+export type GeminiConfig = z.infer<typeof GeminiConfigSchema>;
+
+export const GeminiTtsConfigSchema = GeminiConfigSchema.extend({
+  speechConfig: z
+    .object({
+      voiceConfig: VoiceConfigSchema.optional(),
+      multiSpeakerVoiceConfig: z
+        .object({
+          speakerVoiceConfigs: z
+            .array(
+              z
+                .object({
+                  speaker: z.string().describe('Name of the speaker to use'),
+                  voiceConfig: VoiceConfigSchema,
+                })
+                .describe(
+                  'Configuration for a single speaker in a multi speaker setup'
+                )
+                .passthrough()
+            )
+            .describe('Configuration for all the enabled speaker voices'),
+        })
+        .describe('Configuration for multi-speaker setup')
+        .passthrough()
+        .optional(),
+    })
+    .describe('Speech generation config')
+    .passthrough()
+    .optional(),
+}).passthrough();
+
+// For commonRef
+type ConfigSchema = typeof GeminiConfigSchema | typeof GeminiTtsConfigSchema;
+
+const TTS_MODEL_INFO: ModelInfo = {
+  supports: {
+    multiturn: false,
+    media: false,
+    tools: false,
+    toolChoice: false,
+    systemRole: false,
+    constrained: 'no-tools',
+  },
+};
+
+function commonRef(
+  name: string,
+  info?: ModelInfo,
+  configSchema: ConfigSchema = GeminiConfigSchema
+): ModelReference<ConfigSchema> {
+  return modelRef({
+    name: `googleai/${name}`,
+    info: {
+      supports: {
+        multiturn: info?.supports?.multiturn ?? true,
+        media: info?.supports?.media ?? true,
+        tools: info?.supports?.tools ?? true,
+        toolChoice: info?.supports?.toolChoice ?? true,
+        systemRole: info?.supports?.systemRole ?? true,
+        constrained: info?.supports?.constrained ?? 'no-tools',
+      },
+    },
```

Comment:
```suggestion
    info: info ?? {
      supports: {
        multiturn: true,
        media: true,
        tools: true,
        toolChoice: true,
        systemRole: true,
        constrained: 'no-tools',
      },
    },
```

Reply:
The way I have it you can specify a single thing as non-default, like { supports: { constrained: 'foo' }} and still get the defaults for everything else. Do we not want that?

Reply:
no, when explicitly specified it must include the whole thing. Easier to reason about what the final values are going to be.

Reply:
done